sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -599,6 +599,11 @@ fromClause
     : FROM relation (',' relation)* lateralView* pivotClause?
     ;
 
+temporalClause
+    : FOR? (SYSTEM_VERSION | VERSION) AS OF version=(INTEGER_VALUE | STRING)
```

Comment:
do you mean `( (FOR SYSTEM_VERSION) | VERSION)`?

Reply:
@huaxingao, I don't this change is a good idea. This is only going to cause problems when a user attempts to use `FOR VERSION AS OF` or `SYSTEM_VERSION AS OF`. There's no value in failing queries like that. If we're going to support `VERSION` in place of `SYSTEM_VERSION`, then we should be careful to avoid mistakes like this that result in parse errors and unhappy users.

Reply:
I see, so we wanna unify the Delta lake syntax and SQL Server syntax, in a more organic way. I'm fine with it.

Reply:
Yeah, being needlessly strict is just going to frustrate users.

Reply:
I will submit a follow up PR shortly.